cmd/otk-resolve-containers/main.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/osbuild/images/pkg/blueprint"
+	"github.com/osbuild/images/pkg/container"
+)
+
+// All otk external inputs are nested under a top-level "tree"
+type Tree struct {
+	Tree Input `json:"tree"`
+}
+
+// Input represents the user-provided inputs that will be used to resolve a
+// container image.
+type Input struct {
+	// The architecture of the container images to resolve.
+	Arch string `json:"arch"`
+
+	// List of container refs to resolve.
+	Containers []blueprint.Container `json:"containers"`
+}
+
+// Output contains everything needed to write a manifest that requires pulling
+// container images.
+type Output struct {
+	Const OutputConst `json:"const"`
+}
+
+type OutputConst struct {
+	Containers []ContainerInfo `json:"containers"`
+}
+
+type ContainerInfo struct {
+	Source string `json:"source"`
+
+	// Digest of the manifest at the source.
+	Digest string `json:"digest"`
+
+	// Container image identifier.
+	ImageID string `json:"imageid"`
+
+	// Name to use inside the image.
+	LocalName string `json:"local-name"`
+
+	// Digest of the list manifest at the source
+	ListDigest string `json:"list-digest,omitempty"`
+
+	// The architecture of the image
+	Arch string `json:"arch"`
+
+	LocalStorage bool  `json:"local-storage,omitempty"`
```

Comment:
This means that "LocalStorage: false" will not be part of the json output, not sure that is desirable or now, we could consider not exposing it at all as I'm not sure how we would use it in an otk file or just remove the omitempty(?)

Reply:
I think you're right.  We should probably hide it completely and see if we ever need it.

Reply:
Dropped!